bin/plugin/commands/performance.js

```diff
@@ -162,80 +211,87 @@ async function runPerformanceTests( branches, options ) {
 			'npm run build:packages',
 		].join( ' && ' ) }"`,
-		performanceTestDirectory
+		testRunnerDir
 	);
-	log( '    >> Creating the environment folders' );
-	await runShellScript( 'mkdir -p ' + rootDirectory + '/envs' );
 
-	/*
-	 * 2- Preparing the environment directories per branch.
-	 */
+	logStep( 1, 'Setting up test environments' );
+
+	const envsDir = path.join( baseDir, 'environments' );
+	logStep( 2, 'Creating parent directory:', fAccent( envsDir ) );
+	fs.mkdirSync( envsDir );
+
+	let wpZipUrl = null;
+	if ( options.wpVersion ) {
+		// In order to match the topology of ZIP files at wp.org, remap .0
+		// patch versions to major versions:
+		//
+		//     5.7   -> 5.7   (unchanged)
+		//     5.7.0 -> 5.7   (changed)
+		//     5.7.2 -> 5.7.2 (unchanged)
+		const zipVersion = options.wpVersion.replace( /^(\d+\.\d+).0/, '$1' );
+		wpZipUrl = `https://wordpress.org/wordpress-${ zipVersion }.zip`;
+	}
 
-	log( '\n>> Preparing an environment directory per branch' );
-	const branchDirectories = {};
+	const branchDirs = {};
 	for ( const branch of branches ) {
-		log( `    >> Branch: ${ branch }` );
-		const sanitizedBranch = sanitizeBranchName( branch );
-		const environmentDirectory = rootDirectory + '/envs/' + sanitizedBranch;
+		logStep( 2, 'Branch:', fAccent( branch ) );
+		const sanitizedBranchName = sanitizeBranchName( branch );
+		const envDir = path.join( envsDir, sanitizedBranchName );
+
+		logStep( 3, 'Creating directory:', fAccent( envDir ) );
+		fs.mkdirSync( envDir );
 		// @ts-ignore
-		branchDirectories[ branch ] = environmentDirectory;
-		const buildPath = `${ environmentDirectory }/plugin`;
-		await runShellScript( 'mkdir ' + environmentDirectory );
-		await runShellScript( `cp -R ${ baseDirectory } ${ buildPath }` );
+		branchDirs[ branch ] = envDir;
+		const buildDir = path.join( envDir, 'plugin' );
 
-		const fancyBranch = formats.success( branch );
+		logStep( 3, 'Copying source to:', fAccent( buildDir ) );
+		await runShellScript( `cp -R ${ sourceDir } ${ buildDir }` );
 
-		if ( branch === options.testsBranch ) {
-			log(
-				`        >> Re-using the testing branch for ${ fancyBranch }`
-			);
-			await runShellScript(
-				`cp -R ${ performanceTestDirectory } ${ buildPath }`
-			);
-		} else {
-			log( `        >> Fetching the ${ fancyBranch } branch` );
-			// @ts-ignore
-			await SimpleGit( buildPath ).reset( 'hard' ).checkout( branch );
-		}
+		logStep( 3, 'Checking out:', fAccent( branch ) );
+		// @ts-ignore
+		await SimpleGit( buildDir ).raw( 'checkout', branch );
 
-		log( `        >> Building the ${ fancyBranch } branch` );
+		logStep( 3, 'Installing dependencies and building' );
 		await runShellScript(
-			'bash -c "source $HOME/.nvm/nvm.sh && nvm install && npm ci && npm run prebuild:packages && node ./bin/packages/build.js && npx wp-scripts build"',
-			buildPath
+			`bash -c "${ [
+				'source $HOME/.nvm/nvm.sh',
+				'nvm install',
+				'npm ci',
+				'npm run build',
+			].join( ' && ' ) }"`,
```

Comment:
a change like this could make it harder to find the corresponding source line. probably won't, but we're making a switch from something we can find with a literal string search that's built at compile time to something we're constructing at runtime which combines commands that are more likely to exist in multiple places. personally again, I don't find this change clearer because now I have to manually recombine the pieces the way I expect JS to, and I'm one step further distanced from seeing the raw shell command as it's run. it doesn't help that we're calling BASH and mixing JS strings and JS escaping with the need to understand BASH strings and BASH escaping. is that `'` part of JS or BASH?

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/54309/commits/52d04961ca96f2dd94dec02080a4cfc227a3e0a1